ios/sensor/ReanimatedSensor.m

```diff
@@ -49,7 +53,7 @@ - (bool)initializeGyroscope
                       return;
                     }
                     double data[] = {sensorData.rotationRate.x, sensorData.rotationRate.y, sensorData.rotationRate.z};
-                    self->_setter(data);
+                    self->_setter(data, [self getInterfaceOrientation]);
```

Comment:
How expensive is to call `[self getInterfaceOrientation]`? Should we memoize the value within one frame? Is there any way to keep it local and register some listener to change the value once it changes?

Reply:
There are two ways to use the listener:
- add listener for `UIDeviceOrientationDidChangeNotification` notification, but this is *device* orientation, not *interface* orientation. Device orientation can be changed without changing interface orientation (put phone on flat surface or turn the phone with orientation locked). Or vice-versa (have phone in landscape orientation and lock orientation, it will turn into portrait without moving the device). This approach will be bugged on the second case.
- override `viewWillTransitionToSize` method on some view controller but we don't have any view controller here so I haven't even tested that.

Memoize the value within one frame? We're calling `getInterfaceOrientation` only once per sensor.

As for how expensive this call - it looks like it's just reading a property but who knows, it's not open source 🤷‍♀️ I can try checking in the profiler if you insist.

Reply:
Thanks for clarifying. Since we call it once per sensor and usually there's just one sensor, we don't really need to memoize it. Let's assume that it's blazingly fast, no need to profile it.